Source/Core/Core/HW/EXI/BBA/BuiltIn.cpp

```diff
@@ -488,6 +491,14 @@ void CEXIETHERNET::BuiltInBBAInterface::HandleUDPFrame(const Common::UDPPacket&
   ref->udp_socket.send(data.data(), data.size(), target, ntohs(udp_header.destination_port));
 }
 
+const Common::MACAddress& CEXIETHERNET::BuiltInBBAInterface::ResolveAddress(u32 inet_ip)
+{
+  auto it = m_arp_table.find(inet_ip);
```

Comment:
Could this be replaced with:

```
return m_arp_table.try_emplace(inet_ip, Common::GenerateMacAddress(Common::MACConsumer::BBA)}).first;
```

Reply:
I don't think this approach is suitable since it will constantly generate a new mac address even if it's not used.

Reply:
True, I guess since it is a function it will still be generated, unlike construction.

In that case, maybe use a `hint` to avoid the double map traversal:  https://stackoverflow.com/questions/41507671/what-is-the-use-of-emplace-hint-in-map

Reply:
I'm sorry to be nitpicky at this point but does `find` work?  That SO post seems to indicate you need to use `lower_bound`:

```
auto it = m_arp_table.lower_bound(inet_ip);
if (it != m_arp_table.end() && it->first == inet_ip)
{
  return it->second;
}
else
{
  return m_arp_table.emplace_hint(it, inet_ip, Common::GenerateMacAddress(Common::MACConsumer::BBA)).second;
}
```

Reply:
That's just an optimization so it doesn't have to do a second lookup for the emplace -- the lower_bound already returns the correct position for the insert in the non-matching case. Functionally it's equivalent.

Reply:
> That's just an optimization so it doesn't have to do a second lookup for the emplace

That was actually what sparked this conversation.  Avoiding the second lookup.

Reply:
Oh, I see what you mean. Yes, `emplace_hint` is useless if you pass it the result from `find()`.